lib/collection/src/shards/shard_holder.rs

```diff
@@ -67,26 +70,63 @@ pub enum ShardHashRing {
 }
 
 impl ShardHashRing {
```

Comment:
Maybe a good time to move to a separate file

Reply:
Do you mind if I move this into `collection/src/hash_ring.rs` and rename `HashRing` (that is in `hash_ring.rs` already) into something like `RawOrFair` or just `Inner`?

Reply:
Done.